src/wp-includes/script-loader.php

```diff
@@ -281,10 +281,6 @@ function wp_default_packages_scripts( $scripts ) {
 	 */
 	$assets = include ABSPATH . WPINC . "/assets/script-loader-packages{$suffix}.php";
 
-	// Add the private version of the Interactivity API manually.
```

Comment:
I hope we can safely remove it fully as it wasn’t very hard to use private version of Interactivity API. Technically speaking, folks were able to list is as a dependency so this needs to be investigated carefully.

Reply:
You can keep the script handle for backward compatibility and skip the file reference this way:

https://github.com/WordPress/wordpress-develop/blob/ce84e1f5def19ff6e81c5ef452b8518b8bf03efc/src/wp-includes/script-loader.php#L864

Reply:
The simple way to test it, would be to see what happens when a script defines `wp-interactivity-api` as a dependency with the changes applied. Does it still enqueue?

Reply:
I'll test it. 

To be clear: with the changes in the current PR, I expect that it should not enqueue the Interactivity API. Are you saying that it **should** still enqueue the it (for backward compat)?

Reply:
No, I’m only talking about verifying that scripts that depend on the wp-interactivity-api script handle continue to enqueue. The example presents how to define a script handle without any file on the disk referenced.

Reply:
I recorded a video to illustrate the potential problem better:


https://github.com/WordPress/wordpress-develop/assets/699132/21a28ed3-7751-4947-8710-f3c9fc02aed1

Reply:
ok, I've just tested it and it was never possible for scripts to depend on `wp-interactivity` in the first place because `wp-interactivity` is an ES module. 

I tried enqueuing a script like:

```php
function my_custom_scripts() {
    wp_register_script(
        'my-custom-script',
        includes_url() . '/js/my-custom-script.js',
        array('wp-interactivity'),
        '1.0.0', 
        true
    );
    wp_enqueue_script('my-custom-script');
}

add_action('wp_enqueue_scripts', 'my_custom_scripts');
```

it will enqueue both `my-custom-script` & `wp-interactivity` as scripts (not script modules) which results in:

`Uncaught SyntaxError: Unexpected token 'export' (at index.min.js?ver=1a0bf353a1bf8c59d31b:2070:1)`

Reply:
It doesn't look like the private version of Interactivity API was exposed as ES Module in WordPress 6.4:


https://github.com/WordPress/WordPress/blob/6.4-branch/wp-includes/js/dist/interactivity.js